parsers/parsers.go

```diff
@@ -36,6 +45,35 @@ func (e *Empty) DoFromHuman(string) string {
 	return "Not Yet Implemented"
 }
 
-var ErrTooLarge error = errors.New("Too Beaucoup")
-var ErrTooSmall error = errors.New("Number too small")
-var ErrNotANumber error = errors.New("Not a Number")
+// IsMachineNumber validates that a number:
+// Contains only digits
+// The first digit must be 1-9 when >= 10
+func isMachineNumber(s string) bool {
+	match, _ := regexp.MatchString(`^[0-9]$|^[1-9][0-9]+$`, s)
+	if match {
+		return true
+	}
+	return false
+}
+
+// IsDelimitednumber validates that a number:
+// Contains only digits and` delimiters [., _]
+// The first digit must be 1-9 when >= 10
+// The first group can be 1-3 digits
+// Subsequent groups must be 3 digits
+func isDelimitedNumber(s string) bool {
+	// 0-999 are machine numbers
+	if len(s) < 4 {
+		if ok := isMachineNumber(s); ok {
+			return true
+		} else {
+			return false
+		}
+	}
```

Comment:
I like to keep the  "line of sight" straight and as to the left as possible. Meaning that we should try to reduce the amount of time we have to spend reading `else` branches... visually ... again this has nothing to do with the code correctness and more of a nicety for the reader.

So for the first check we can optimize for visibility by only caring about the early exit, in this case we really only care about the case of "true" because once we have that we're done

```suggestion
	if len(s) < 4 && isMachineNumber(s) {
			return true    
	}
```

we can ignore the `false` case because the next check will take care of that for us.